app/models/pet.rb

```diff
@@ -75,6 +77,28 @@ def append_images=(attachables)
     images.attach(attachables)
   end
 
+  # This will be approximate for multiple reasons
+  # - we aren't checking whether today's date has passed the pet's birthdate but instead calculating from duration
+  # - most birthdays entered in the system aren't exact
+  def approximate_age
+    today_in_seconds_since_unix_epoch = DateTime.current.to_i
+    pet_birth_date_in_seconds_since_unix_epoch = self.birth_date.to_i
+    pet_life_duration = ActiveSupport::Duration.build(today_in_seconds_since_unix_epoch - pet_birth_date_in_seconds_since_unix_epoch)
+
+    # years and months will be missing if they are 0, so initialize hash with 0 values and merge in the computed ones, if any
+    {years: 0, months: 0}.merge(pet_life_duration.parts.slice(:years, :months))
+  end
+
+  def approximate_age_display
+    approximate_age => {years:, months:}
+
+    years_string = years.positive? ? "#{years} years" : nil
+    months_string = months.positive? ? "#{months} months" : nil
+
+    # using compact and join will correctly handle the case when either years or months is 0
+    [years_string, months_string].compact.join(', ')
+  end
```

Comment:
@edwinthinks suggested IRL to use https://api.rubyonrails.org/classes/ActionView/Helpers/DateHelper.html#method-i-distance_of_time_in_words, which doesn't produce the exact same output, but since this is all approximate anyway I think it's worth switching to it